src/cart/crt_init.c

```diff
@@ -198,13 +197,20 @@ prov_data_init(struct crt_prov_gdata *prov_data, crt_provider_t provider,
 	if (rc != 0)
 		return rc;
 
-	/* Set max number of contexts. Defaults to the number of cores */
-	ctx_num = 0;
-	d_getenv_uint("CRT_CTX_NUM", &ctx_num);
-	if (opt)
-		max_num_ctx = ctx_num ? ctx_num : max(crt_gdata.cg_num_cores, opt->cio_ctx_max_num);
-	else
-		max_num_ctx = ctx_num ? ctx_num : crt_gdata.cg_num_cores;
+	if (crt_is_service()) {
```

Comment:
in a current test (test_daos_degraded_mode) it appears to have set CRT_CTX_SHARE_ADDR = 1 and CRT_CTX_NUM = 8 which (after a recent master commit) causes engine startup errors, but previously worked.

With this change, it looks like the test environment's setting of CRT_CTX_SHARE_ADDR will be ignored, and the CRT_CTX_NUM value will also be ignored, and instead be set to CRT_SRV_CONTEXT_NUM(=64)? 

Ok, actually in separate chat with @frostedcmos previously when the test worked the share addr (SEP) and CTX_NUM settings were ignored in engine anyway. So this change gets the engine setup the way we had it before when the testing worked?

Reply:
Correct. Previous setting of CRT_CTX_NUM=8 would be ignored on the engines due to SEP being unsupported by providers, so this patch should bring engines back to the same behavior as before my patch that broke things.